### PR TITLE
Introduce TypeScript and official DevTools Protocol typings

### PR DESCRIPTION
**Pull request by max-vogler**
This PR introduces TypeScript to double down on the existing effort on introducing typings:
- Type information is now checked at compile time.
- Official typings from https://github.com/ChromeDevTools/devtools-protocol are used to provide better correctness guarantees for Debugger method names, event names, and event interfaces. These replace the previous, handwritten types.
- Source maps are emitted during development build `npm run dev` to allow inspecting the code in DevTools.

These changes do not affect the extension bundle size: It stayed at 179k.

TODO:
- [x] Do not generate source maps in production build.

### CODE REVIEW

File: src/devtools/WebAudioGraphIntegrator.ts [hunk 1]
```diff
@@ -0,0 +1,213 @@
+import dagre from 'dagre';
```
Comment (hoch) [inline src/devtools/WebAudioGraphIntegrator.ts:1]: Since this is a new file, can we add 1) a license and 2) a top-level overview?

**Review by hoch — CHANGES_REQUESTED** [review]
This change looks neat!

File: src/chrome/DebuggerWebAudioDomain.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import {ProtocolMapping} from 'devtools-protocol/types/protocol-mapping';
+
+export const Methods: {[K in string]: keyof ProtocolMapping.Commands & `WebAudio.${K}`} = {
```
[the remark this answers is not in the document]
Comment (max-vogler) [reply src/chrome/DebuggerWebAudioDomain.ts:3]: Type parameters like `K` are often times single-digit names, similar to Java Generics. I picked `K` as shorthand for key.

File: src/devtools/WebAudioGraphIntegrator.ts (hunk 1 above)
Comment (max-vogler) [reply src/devtools/WebAudioGraphIntegrator.ts:1]: git's diff sadly is not powerful enough to show that WebAudioGraphIntegrator.ts is actually renamed/branched from WebAudioGraphIntegrator.js

Here's a screenshot to give you an slight impression of before and after:
image

Comment (max-vogler) [reply src/devtools/WebAudioGraphIntegrator.ts:1]: I suggest to add license headers for all files at once in a future PR.

**Pull request merged**